templates/Dockerfile.tpl

```diff
@@ -0,0 +1,31 @@
+{{ template "boilerplate_hash" }}
+
+# Build the manager binary
+FROM golang:1.14.1 as builder
+
+ARG work_dir=/github.com/aws/aws-service-operator-k8s
+WORKDIR $work_dir
+# For building Go Module required
+ENV GOPROXY=direct
+ENV GO111MODULE=on
+ENV GOARCH=amd64
+ENV GOOS=linux
+ENV CGO_ENABLED=0
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN  go mod download
+# Copy the go source
+# TODO: for now copy whole repository. Later once autogeneration is ready, need to determine which dependencies to take in
```

Comment:
Let's change this now. This should be:

```
COPY services/{{ .ServiceAlias }}/. $work_dir
```

Reply:
I wasn't sure what will be consumed until complete implementation is ready, so kept it as TODO. 

As an example, runtime
https://github.com/aws/aws-service-operator-k8s/tree/mvp/pkg/runtime

Also we will have to maintain paths as following:
from source of root we have to copy:
COPY services/{{ .ServiceAlias }}/.  $work_dir/{{ .ServiceAlias }}/.
COPY pkg/runtime $work_dir/pkg/runtime
and more dependencies that we may need.


And Build will become this:
RUN  go build -a -o $work_dir/bin/controller $work_dir/{{ .ServiceAlias }}/cmd/controller/main.go

Reply:
yeah, that makes sense... OK, for now let's keep it like this and refactor later.